File: cadwiki-nuget/cadwiki.DllReloader/AutoCAD/UiRibbon/Buttons/UiRouter.vb
Option Strict On
Option Infer Off
Option Explicit On
Imports System.Reflection

Namespace AutoCAD.UiRibbon.Buttons
    Public Class UiRouter
        Public FullClassName As String
        Public MethodName As String
        Public Parameters As Object()
        Public AssemblyName As String
        Public NetReloader As AutoCADAppDomainDllReloader
        Public IExtensionAppAssembly As Assembly

        Public Sub New(fullClassName As String, methodName As String, parameters As Object(),
                       netReloader As AutoCADAppDomainDllReloader, iExtensionAppAssembly As Assembly)
            Me.FullClassName = fullClassName
            Me.MethodName = methodName
            Me.Parameters = parameters
            Me.NetReloader = netReloader
            Me.IExtensionAppAssembly = iExtensionAppAssembly
        End Sub
    End Class
End Namespace

File: cadwiki-nuget/cadwiki.DllReloader/AutoCAD/UiRibbon/Buttons/UiRouter.cs
using System.Reflection;

namespace cadwiki.DllReloader.AutoCAD.UiRibbon.Buttons
{
    public class UiRouter
    {
        public string FullClassName;
        public string MethodName;
        public object[] Parameters;
        public string AssemblyName;
        public AutoCADAppDomainDllReloader NetReloader;
        public Assembly IExtensionAppAssembly;

        public UiRouter(string assemblyName, string fullClassName, string methodName, object[] parameters, AutoCADAppDomainDllReloader netReloader, Assembly iExtensionAppAssembly)
        {
            AssemblyName = assemblyName;
            FullClassName = fullClassName;
            MethodName = methodName;
            Parameters = parameters;
            NetReloader = netReloader;
            IExtensionAppAssembly = iExtensionAppAssembly;
        }
    }
}